src/main/haskell/kore/src/Kore/Step/Simplification/Data.hs

```diff
@@ -28,33 +31,41 @@ simplification of a MetaMLPattern was correct.
 data SimplificationProof level = SimplificationProof
     deriving (Show, Eq)
 
-type Simplifier = Counter
+type Simplifier = ReaderT Integer Counter
```

Comment:
I would have added it to `MetadataTools`... If you think that's not the right place (for sure it's not perfect), you should, at least, wrap the `Integer` in a type with `SmtTimeout` in its name, because people forget to document the meaning of plain types like `Integer` when using them as arguments or as part of other types.

Reply:
I agree that the time-out value should be a `newtype SmtTimeOut`, but I don't agree with adding it to `MetadataTools`. The time-out doesn't have anything to do with the `IndexedModule`. `MetadataTools` should not just be a catch-all.